web/sites/default/settings.php

```diff
@@ -2,264 +2,80 @@
-    'driver' => 'mysql',
-    'database' => getenv('MARIADB_DATABASE') ?: 'drupal',
-    'username' => getenv('MARIADB_USERNAME') ?: 'drupal',
-    'password' => getenv('MARIADB_PASSWORD') ?: 'drupal',
-    'host' => getenv('MARIADB_HOST') ?: 'mariadb',
-    'port' => 3306,
-    'charset' => 'utf8mb4',
-    'collation' => 'utf8mb4_general_ci',
-  ];
-}
-
-// Lagoon Solr connection
-if (getenv('LAGOON')) {
-  $config['search_api.server']['backend_config']['connector_config']['host'] = getenv('SOLR_HOST') ?: 'solr';
-  $config['search_api.server']['backend_config']['connector_config']['path'] = '/solr/' . getenv('SOLR_CORE') ?: 'drupal';
-}
-
-// Lagoon Varnish & reverse proxy settings
-if (getenv('LAGOON')) {
-  $varnish_control_port = getenv('VARNISH_CONTROL_PORT') ?: '6082';
-  $varnish_hosts = explode(',', getenv('VARNISH_HOSTS') ?: 'varnish');
-  array_walk($varnish_hosts, function (&$value, $key) use ($varnish_control_port) {
-    $value .= ":$varnish_control_port";
-  });
-
-  $settings['reverse_proxy'] = TRUE;
-  $settings['reverse_proxy_addresses'] = array_merge(explode(',', getenv('VARNISH_HOSTS')), ['varnish']);
-  $settings['varnish_control_terminal'] = implode($varnish_hosts, " ");
-  $settings['varnish_control_key'] = getenv('VARNISH_SECRET') ?: 'lagoon_default_secret';
-  $settings['varnish_version'] = 4;
-}
-
-// Redis configuration.
-if (getenv('LAGOON') && (getenv('ENABLE_REDIS'))) {
-  $settings['redis.connection']['interface'] = 'PhpRedis';
-  $settings['redis.connection']['host'] = getenv('REDIS_HOST') ?: 'redis';
-  $settings['redis.connection']['port'] = 6379;
-
-  $settings['cache_prefix']['default'] = getenv('LAGOON_PROJECT') . '_' . getenv('LAGOON_GIT_SAFE_BRANCH');
+// Set by Drupal core.
+/* @var $app_root string */
 
-  # Do not set the cache during installations of Drupal
-  if (!drupal_installation_attempted()) {
-    $settings['cache']['default'] = 'cache.backend.redis';
+// Path to standardized platform-wide settings includes.
+/* @var $govcms_includes string */
 
-    // Include the default example.services.yml from the module, which will
-    // replace all supported backend services (that currently includes the cache tags
-    // checksum service and the lock backends, check the file for the current list)
-    $settings['container_yamls'][] = 'modules/contrib/redis/example.services.yml';
+// Set to 'local', 'development', 'production'.
+/* @var $govcms_env string */
 
-    // Allow the services to work before the Redis module itself is enabled.
-    $settings['container_yamls'][] = 'modules/contrib/redis/redis.services.yml';
+$govcms_includes = $app_root . '/../vendor/govcms/scaffold-tooling/drupal/settings';
 
-    // Manually add the classloader path, this is required for the container cache bin definition below
-    // and allows to use it without the redis module being enabled.
-    $class_loader->addPsr4('Drupal\\redis\\', 'modules/contrib/redis/src');
+// Todo, this logic to be reviewed by Amazee.
```

Comment:
```
// @todo:
```

Reply:
> 1. Is it possible to include contrib drupal modules as satis is now used instead of drupal.org?

Yes PaaS will be allowed to add the required packagist references. Whether this is done as an Ansible/AWX process is not clear yet, but it's a pretty simple thing to instruct.

> 2. How does the SaaS locking mechanism work now? Previously, every project provisioned in GitLab had `gitlab-ci.yml` locked for SaaS. Could you please expand if this changed.

.gitlab-ci.yml remains locked. Even for PaaS I'd plan to recommend forking scaffold-tooling, and only changing that reference. But PaaS can do anything of course. For everyone else it's very locked as otherwise we'll not be able to ensure clients can't disable vetting(static) and auditing(drutiny) in CI.

> 3. Is it possible to create a release of tooling repo and reference the version in `composer.json` instead of using `master-dev`.

That will happen in this PR.

> 4. Can this PR have documentation page added (or readme.md updated) to reflect new changes. To be honest, the new approach is cleaner, but not something that majority site builders is used to. If you could copy-paste the description of this PR into readme file it would help heaps.

I would argue that it's much clearer for SaaS, since  current saas is an island, and for PaaS there will be instructions on how they can "normalize" to taste. 

> 5. Has enabling of XDebug been or will be addressed in this PR? It was/is a pain to enable it in current setup.

Nah, I have to keep stuff like this out of scope or it would be too much to get in at once. This PR (along with the recent change to enable use of versioned images and test environments in Lagoon), is intended in a variety of ways to make managing the platform much easier to manage and innovate on, hopefully unblocking things like xdebug improvements.

> 6. Re not running ahoy commands in CI - this makes sense if ahoy commands have no logic because everything now is in scripts. However, we discussed this separately, that having ahoy commands in CI has 2 advantages:

There is a pretty widespread consensus that ahoy commands should just be local. The use of shared ahoy commands locally and in CI conflates the context and intention of those commands, and I can point to a number of places where they are being "overloaded" with workarounds for those different contexts. The current .ahoy commands need a good work-over, and this is made very hard by their use in CI, so we are slowly removing them from CI usage. I think the ideal of using them universally is great, but the application is not feasible in a a project like this.

Reply:
@simesy I provided 2 advantage examples for using ahoy commands in CI and they both are valid. 

Saying " pretty widespread consensus that ahoy commands should just be local" does not have any explanation about WHY there is this consensus.

I will leave this alone, but I think this is a lost opportunity to let other people to understand how CI works and that CI is not that magical.

Reply:
Sorry by "consensus" I mean we have quite a few people on deck at the moment and the decision was unanimous. GitLab and local environments operate in subtly different ways, and we so we want to handle the idiosyncrasies in different places.